sdk/dotnet/Pulumi/Deployment/GrpcMonitor.cs

```diff
@@ -1,22 +1,37 @@
 // Copyright 2016-2020, Pulumi Corporation
 
-using System.Collections.Generic;
+using System;
 using System.Threading.Tasks;
-using Grpc.Core;
+using Grpc.Net.Client;
 using Pulumirpc;
+using Grpc.Core;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
 
 namespace Pulumi
 {
     internal class GrpcMonitor : IMonitor
     {
         private readonly ResourceMonitor.ResourceMonitorClient _client;
-
+        private static ConcurrentDictionary<string, GrpcChannel> _monitorChannels = new ConcurrentDictionary<string, GrpcChannel>();
         public GrpcMonitor(string monitor)
         {
             // maxRpcMessageSize raises the gRPC Max Message size from `4194304` (4mb) to `419430400` (400mb)
             var maxRpcMessageSize = 400 * 1024 * 1024;
-            var grpcChannelOptions = new List<ChannelOption> { new ChannelOption(ChannelOptions.MaxReceiveMessageLength, maxRpcMessageSize)};
-            this._client = new ResourceMonitor.ResourceMonitorClient(new Channel(monitor, ChannelCredentials.Insecure, grpcChannelOptions));
+            if (!_monitorChannels.ContainsKey(monitor))
```

Comment:
Wonder if we should have a "GrpcPool" class to share between GrpcEngine and GrpcMonitor.

Reply:
i think for this use case it is simple enough to keep it as imo